tests/unit/test_internal_utils.py

```diff
@@ -803,3 +804,75 @@ def test_get_dataset_type__conflicting_data():
         else:
             with pytest.raises(PowerGridError):
                 get_dataset_type(data=data)
+
+
+@pytest.fixture
+def row_dense_data():
+    source = initialize_array(DatasetType.update, ComponentType.source, (2, 3))
+    source["id"] = [[0, 2, 3], [0, 2, 3]]
+    source["u_ref"] = [[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]]
+
+    sym_load = initialize_array(DatasetType.update, ComponentType.sym_load, (2, 1))
+    sym_load["id"] = [[1], [1]]
+    sym_load["p_specified"] = [[100.0], [200.0]]
+
+    return {
+        ComponentType.source: source,
+        ComponentType.sym_load: sym_load,
+    }
+
+
+@pytest.fixture
+def row_sparse_data():
+    transformer = initialize_array(DatasetType.update, ComponentType.transformer, 1)
+    transformer["id"] = 1
+    transformer["tap_pos"] = 3
+
+    sym_gen = initialize_array(DatasetType.update, ComponentType.sym_gen, (2, 4))
```

Comment:
Sparse data has a SingleArray under data. ie. `sym_gen = initialize_array(DatasetType.update, ComponentType.sym_gen, 8)`

Reply:
Can you add an exception for that too under https://github.com/PowerGridModel/power-grid-model/blob/main/src/power_grid_model/_utils.py#L420